test/fft_utest.cpp

```diff
@@ -483,3 +548,204 @@ TYPED_TEST(FftTest, TestFftGf2n) // NOLINT
+    _a1 = std::make_unique<quadiron::vec::Vector<TypeParam>>(*gf, n);
+    _a2 = std::make_unique<quadiron::vec::Vector<TypeParam>>(*gf, n);
+    a->rand();
+
+    fft->fft(*_a1, *a);
+    fft2->fft(*_a2, *a);
+
+    ASSERT_EQ(*_a1 == *_a2, true);
+}
+
+TYPED_TEST(FftTest, TestFftEquivalence) // NOLINT
+{
+    std::vector<uint64_t> qs({7681, 12289, 65537});
+    std::vector<uint64_t> ns({256, 512, 1024});
+
+    for (std::vector<uint64_t>::iterator it = qs.begin(); it != qs.end();
+         it++) {
+        TypeParam q = *it;
+
+        for (std::vector<uint64_t>::iterator it2 = ns.begin(); it2 != ns.end();
+             it2++) {
+            TypeParam n = *it2;
+
+            ASSERT_EQ(quadiron::arith::is_prime(q), true);
+
+            std::unique_ptr<gf::Field<TypeParam>> gf = nullptr;
+            std::unique_ptr<quadiron::vec::Vector<TypeParam>> a = nullptr;
+            std::unique_ptr<quadiron::vec::Vector<TypeParam>> _a1 = nullptr;
+            std::unique_ptr<quadiron::vec::Vector<TypeParam>> _a2 = nullptr;
+            std::unique_ptr<quadiron::vec::Vector<TypeParam>> _a3 = nullptr;
+            std::unique_ptr<fft::FourierTransform<TypeParam>> fft = nullptr;
+            std::unique_ptr<fft::FourierTransform<TypeParam>> fft2 = nullptr;
+            std::unique_ptr<fft::FourierTransform<TypeParam>> fft3 = nullptr;
+
+            gf = gf::alloc<gf::Field<TypeParam>, gf::Prime<TypeParam>>(q);
+
+            TypeParam r = gf->get_nth_root(n);
+            int l = arith::log2<TypeParam>(n);
+            fft = std::make_unique<fft::Naive<TypeParam>>(*gf, n, r);
+            try {
+                fft2 = std::make_unique<fft::Large<TypeParam>>(*gf, l, r);
+            } catch (quadiron::Exception&) {
+                fft2 = nullptr;
+            }
+            try {
+                fft3 =
+                    std::make_unique<fft::Radix2<TypeParam>>(*gf, n, n, 1024);
+            } catch (quadiron::Exception&) {
+                fft3 = nullptr;
+            }
+
+            a = std::make_unique<quadiron::vec::Vector<TypeParam>>(*gf, n);
+            _a1 = std::make_unique<quadiron::vec::Vector<TypeParam>>(*gf, n);
+            _a2 = std::make_unique<quadiron::vec::Vector<TypeParam>>(*gf, n);
+            _a3 = std::make_unique<quadiron::vec::Vector<TypeParam>>(*gf, n);
+            a->rand();
+
+            fft->fft(*_a1, *a);
+            if (fft2) {
+                fft2->fft(*_a2, *a);
+            }
+            if (fft3) {
+                fft3->fft(*_a3, *a);
+            }
+            if (fft2) {
+                ASSERT_EQ(*_a1 == *_a2, true);
+            }
+            if (fft3) {
+                ASSERT_EQ(*_a1 == *_a3, true);
+            }
+        }
+    }
+}
+
+TYPED_TEST(FftTest, TestFftLarge2) // NOLINT
+{
+    // Schönhage-Strassen algorithm
+    // Example taken from Pierre Meunier's book
+    if (sizeof(TypeParam) < 8)
+        return;
+
+    int b = 10; // base
```

Comment:
If you need a comment explaining `b` is 'base', consider calling the variable `base` instead of `b`.